### PR TITLE
Feature/typescript migration 2

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Main Changes:
* Continuing the conversion of the codebase to TypeScript.

### CODE REVIEW

**Comment by cschwinderg** [comment]
@maxdiebold-erg Just starting to look at this. When you get a chance, can you fix the merge conflicts.

**Comment by maxdiebold-erg** [comment]
@cschwinderg The merge conflicts should be fixed now. In the process, I found a DOM nesting issue with the tooltips, so I altered that a bit on the Monitoring Report page, too.

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Thank you for working on this. Looks great! Glad to see us moving more towards TypeScript. 

I did find a bug and some other small items (listed below).

File: app/client/src/types/index.ts [hunk 1]
```diff
@@ -195,11 +245,117 @@ export interface UsgsStreamgage {
...
+        value: [
+          {
+            dateTime: string; // ISO format datetime
+            qualifiers: ['P'];
+            value: string; // number
+          },
+        ];
+      }[];
+      variable: {
+        noDataValue: -999999;
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/types/index.ts:296]: True, don't know why I made that assumption.

File: app/client/src/utils/mapFunctions.tsx [hunk 2]
```diff
@@ -502,88 +598,90 @@ const organizationMapping = {
...
   if (!attributes) return 'Waterbody ID:';
 
-  const mappedLabel = organizationMapping[attributes.organizationid];
-  if (mappedLabel) return `${mappedLabel} Waterbody ID:`;
-  if (attributes.orgtype === 'Tribe') return 'Tribal Waterbody ID:';
-  if (attributes.orgtype === 'State') return 'State Waterbody ID:';
+  const { organizationid, orgtype } = attributes;
+  Object.values(organizationMapping).forEach(([label, value]) => {
+    if (label === organizationid) return `${value} Waterbody ID:`;
+  });
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/utils/mapFunctions.tsx:609]: Good catch!

File: app/client/src/components/shared/MapWidgets.tsx [hunk 3]
```diff
@@ -140,34 +151,34 @@ function handleMapZoomChange(view: __esri.View) {
...
-      if (layer.layers) {
+      if (isGroupLayer(layer)) {
         let anyVisible = false;
         layer.layers.forEach((sublayer) => {
           if (sublayer.visible) anyVisible = true;
         });
         if (!anyVisible) return;
       }
-      if (layer.sublayers) {
+      if (isTileLayer(layer) || isMapImageLayer(layer)) {
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/shared/MapWidgets.tsx]: True! I went ahead and made a union type for all Layer subclasses with sublayers.

**Comment by maxdiebold-erg** [comment]
@cschwinderg Good calls, I made the changes you pointed out, let me know if you see any others.

**Review by cschwinderg — APPROVED** [review]
Looks good, thanks for making those changes.

**Pull request merged**